File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Controllers/UnidadeContaController.cs
using JsonWebToken.Code.UnidadeConta;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace JsonWebToken.Model
{
  public class UnidadeContaController : ApiController
  {
    // GET api/<controller>
    public IEnumerable<UnidadeContaModel> Get()
    {
      UnidadeContaBLL bll = new UnidadeContaBLL();
      return bll.SelectUnidadeConta();
    }

    // GET api/<controller>/5
    public string Get(int id)
    {
     // UnidadeContaBLL bll = new UnidadeContaBLL();
      return "value";
    }

    // POST api/<controller>
    public void Post([FromBody]UnidadeContaModel unidadeConta)
    {
      UnidadeContaBLL bll = new UnidadeContaBLL();
     
        bll.InsertUnidadeConta(unidadeConta);
      }

    // PUT api/<controller>/5
    public void Put([FromBody]UnidadeContaModel unidadeConta, string acao)
    {
      UnidadeContaBLL bll = new UnidadeContaBLL();
      if (acao.ToLower().Equals("ativar"))
      {
        bll.AtivarUnidadeConta(unidadeConta);
      }
      else if (acao.ToLower().Equals("inativar"))
      {
        bll.InativarUnidadeConta(unidadeConta);
      }
      
    }

    // DELETE api/<controller>/5
    public void Delete(int id)
    {
      //não realiza esta operacao
      //  UnidadeContaBLL bll = new UnidadeContaBLL();

    }
  }
}

File: Backend/Administracao.ConciliacaoAPI/Controllers/UnidadeContaController.vb
Imports System.Net
Imports System.Net.Http
Imports System.Web.Http
Imports Administracao.ConciliacaoAPI.Code.UnidadeConta
Imports Administracao.ConciliacaoAPI.Model

Namespace Controllers
    Public Class UnidadeContaController
        Inherits ApiController
        ' GET api/<controller>
        Public Function GetValues() As Object
            Try
                Dim bll As New UnidadeContaBLL()
                Return New ResultModel(bll.SelectUnidadeConta())
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' GET api/<controller>/5
        <HttpGet>
        <Route("api/UnidadeConta/{codUnidade}/{codConta}")>
        Public Function GetValues(CodUnidade As Integer, codConta As Integer) As Object
            Try
                Dim bll As New UnidadeContaBLL()
                Return New ResultModel(bll.SelectUnidadeConta(CodUnidade, codConta))
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' POST api/<controller>
        Public Function PostValue(<FromBody> unidadeConta As UnidadeContaModel) As Object
            Try
                Dim bll As New UnidadeContaBLL()
                bll.InsertUnidadeConta(unidadeConta)
                Return New ResultModel(Nothing)
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' PUT api/<controller>/5
        <HttpPut()>
        <Route("api/UnidadeConta/{codEmpresa}/{codConta}/{codUnidade}")>
        Public Function PutValue(codEmpresa As Integer, codConta As Integer, codUnidade As Integer, <FromBody> unidadeConta As UnidadeContaModel) As Object
            Try
                Dim bll As New UnidadeContaBLL()

                unidadeConta = New UnidadeContaModel()
                unidadeConta.ContaModel = New ContaModel()
                unidadeConta.UnidadeModel = New UnidadeModel()
                unidadeConta.CodEmpresa = codEmpresa
                unidadeConta.UnidadeModel.CodUnidade = codUnidade
                unidadeConta.ContaModel.CodContaContabil = codConta

                bll.AtivarInativarUnidadeConta(unidadeConta)

                Return New ResultModel(Nothing)

            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

    End Class
End Namespace
